plone/app/contenttypes/migration/migration.py

```diff
@@ -205,10 +205,15 @@ def last_migrate_layout(self):
         migrate_properties.
         """
         old_layout = self.old.getLayout() or getattr(self.old, 'layout', None)
-        if old_layout in LISTING_VIEW_MAPPING.keys():
+        if old_layout:
             default_page = self.old.getDefaultPage() or \
                 getattr(self.old, 'default_page', None)
-            self.new.setLayout(LISTING_VIEW_MAPPING[old_layout])
+            try:
+                # Delete old-style layout attribute.
+                del self.new.layout
+            except AttributeError:
+                pass
+            self.new.setLayout(LISTING_VIEW_MAPPING.get(old_layout, old_layout))  # noqa
```

Comment:
Here, in any case (as long as old_layout has some value) the layout is set with setLayout. If no LISTING_VIEW_MAPPING is defined, it falls back to the previous layout name.